deepwell/src/services/user/service.rs

```diff
@@ -159,14 +161,57 @@ impl UserService {
             }
         };
 
+        // Perform email verification.
+        //
+        // If the email is either disposable or invalid, propogate the error upwards and
+        // stop the account creation. If the email passes validation, mark if it's an alias
+        // or not.
+        //
+        // The assigned variable is also used to check whether email validation occurred, as it
+        // will always be `Some` if validation occurred and `None` otherwise.
+        let email_is_alias = if !bypass_email_verification {
+            let email_validation_output = EmailService::validate(&email).await?;
+
+            match email_validation_output.classification {
+                EmailClassification::Normal => {
+                    tide::log::info!("User {slug}'s email was verified successfully");
+                    Some(false)
+                }
+
+                EmailClassification::Alias => {
+                    tide::log::info!(
+                        "User {slug}'s email was verified successfully (as an alias)"
+                    );
+                    Some(true)
+                }
+
+                EmailClassification::Disposable => {
+                    tide::log::error!(
+                        "User {slug}'s email is disposable and did not pass verification"
+                    );
+                    return Err(Error::DisallowedEmail);
+                }
+
+                EmailClassification::Invalid => {
+                    tide::log::error!(
+                        "User {slug}'s email is invalid and did not pass verification"
+                    );
+                    return Err(Error::InvalidEmail);
+                }
+            }
+        } else {
+            None
```

Comment:
Since it's a long branch, a comment could be helpful to illustrate what this case is:
```suggestion
            // Skipping email verification
            None
```